src/Core/src/Credentials/ContainerProvider.php

```diff
@@ -0,0 +1,108 @@
+<?php
+
+declare(strict_types=1);
+
+namespace AsyncAws\Core\Credentials;
+
+use AsyncAws\Core\Configuration;
+use Psr\Log\LoggerInterface;
+use Psr\Log\NullLogger;
+use Symfony\Component\HttpClient\Exception\JsonException;
+use Symfony\Component\HttpClient\Exception\TransportException;
+use Symfony\Component\HttpClient\HttpClient;
+use Symfony\Contracts\HttpClient\Exception\DecodingExceptionInterface;
+use Symfony\Contracts\HttpClient\Exception\HttpExceptionInterface;
+use Symfony\Contracts\HttpClient\Exception\TransportExceptionInterface;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+use Symfony\Contracts\HttpClient\ResponseInterface;
+
+/**
+ * Provides Credentials from the running ECS.
+ *
+ * @see https://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/index.html?com/amazonaws/auth/ContainerCredentialsProvider.html
+ */
+final class ContainerProvider implements CredentialProvider
+{
+    private const ENDPOINT = 'http://169.254.170.2';
+    private const ENV_URI = 'AWS_CONTAINER_CREDENTIALS_RELATIVE_URI';
+    private const ENV_TIMEOUT = 'AWS_METADATA_SERVICE_TIMEOUT';
+
+    private $logger;
+
+    private $httpClient;
+
+    private $timeout;
+
+    private $envUri;
+
+    public function __construct(?HttpClientInterface $httpClient = null, ?LoggerInterface $logger = null, float $timeout = 1.0)
+    {
+        $this->logger = $logger ?? new NullLogger();
+        $this->httpClient = $httpClient ?? HttpClient::create();
+        $this->timeout = getenv(self::ENV_TIMEOUT) ?: $timeout;
+        $this->envUri = getenv(self::ENV_URI) ?: '';
+    }
+
+    public function getCredentials(Configuration $configuration): ?Credentials
+    {
+        // fetch credentials from ecs endpoint
+        try {
+            $response = $this->httpClient->request('GET', self::ENDPOINT . $this->envUri, ['timeout' => $this->timeout]);
```

Comment:
`$this->envUri` can be empty, the AWS SDK contains an early exit that "disable" the provider if the env variable is missing.

https://github.com/aws/aws-sdk-php/blob/af91c2fc467a326e5bcb5665e4e2ad3d84d28be2/src/Credentials/CredentialProvider.php#L90-L92